File: backend/components/metiers/administration_publique/utils/rbac/core/rbac_core.test.py
# rbac_core.test.py
"""Tests unitaires core RBAC Python"""
from .rbac_core import check_permission, get_user_roles

def test_check_permission():
    class User: permissions = ['read']
    assert check_permission(User, 'read')
    assert not check_permission(User, 'write')

def test_get_user_roles():
    class User: roles = ['admin']
    assert 'admin' in get_user_roles(User)

File: backend/components/metiers/administration_publique/utils/rbac/core/rbac_core.test.js
// rbac_core.test.js
// Tests unitaires core RBAC JS
const { checkPermission, getUserRoles } = require('./rbac_core');
describe('core RBAC JS', () => {
  it('vérifie la permission', () => {
    expect(checkPermission({permissions:['read']}, 'read')).toBe(true);
    expect(checkPermission({permissions:['read']}, 'write')).toBe(false);
  });
  it('récupère les rôles', () => {
    expect(getUserRoles({roles:['admin']})).toContain('admin');
  });
});
